src/Microsoft.OData.Client/EntityTracker.cs

```diff
@@ -576,4 +561,16 @@ private void ValidateDuplicateIdentity(Uri identity, EntityDescriptor descriptor
             }
         }
     }
+
+    /// <summary>
+    /// An object instance equality comparer
+    /// </summary>
+    /// <typeparam name="TObject"></typeparam>
+    public class EntityEqualityComparer<TObject> : IEqualityComparer<TObject> where TObject : class

```

Comment:
From the docs, they recommend deriving from `EqualityComparer<T>` instead of the `IEqualityComparer<T>`. Is it something you can consider? https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.iequalitycomparer-1?view=net-5.0